logsmanagement/functions.c

```diff
@@ -0,0 +1,1118 @@
+                            sizeof(LOGS_MANAG_FUNC_PARAM_DATA_ONLY ":") - 1)) {
+
+            char *v = &keyword[sizeof(LOGS_MANAG_FUNC_PARAM_DATA_ONLY ":") - 1];
+
+            if(!strcmp(v, "false") || !strcmp(v, "no") || !strcmp(v, "0"))
+                data_only = false;
+            else
+                data_only = true;
+        }
+        // else if(strncmp(keyword, JOURNAL_PARAMETER_SLICE ":", sizeof(JOURNAL_PARAMETER_SLICE ":") - 1) == 0) {
+        //     char *v = &keyword[sizeof(JOURNAL_PARAMETER_SLICE ":") - 1];
+
+        //     if(strcmp(v, "false") == 0 || strcmp(v, "no") == 0 || strcmp(v, "0") == 0)
+        //         slice = false;
+        //     else
+        //         slice = true;
+        // }
+        else if(strncmp(keyword, LOGS_MANAG_FUNC_PARAM_ID ":", sizeof(LOGS_MANAG_FUNC_PARAM_ID ":") - 1) == 0) {
+            char *id = &keyword[sizeof(LOGS_MANAG_FUNC_PARAM_ID ":") - 1];
+
+            if(*id)
+                progress_id = id;
+        }
+
+        else if(strncmp(keyword, LOGS_MANAG_FUNC_PARAM_SOURCE ":", sizeof(LOGS_MANAG_FUNC_PARAM_SOURCE ":") - 1) == 0) {
+            source = !strcmp("all", &keyword[sizeof(LOGS_MANAG_FUNC_PARAM_SOURCE ":") - 1]) ? 
+                NULL : &keyword[sizeof(LOGS_MANAG_FUNC_PARAM_SOURCE ":") - 1];
+        }
+
+
+
+        // char *key = strsep_skip_consecutive_separators(&value, ":");
+        // if(!key || !*key) continue;
+        // if(!value || !*value) continue;
+
+        // Kludge to respect quotes with spaces in between
+        // Not proper fix
+        // if(*value == '_'){
+        //     value++;
+        //     value[strlen(value)] = ' '; 
+        //     value = strsep_skip_consecutive_separators(&value, "_");
+        //     function = strrchr(value, 0);
+        //     function++;
+        // }
+
+        else if(strncmp(keyword, LOGS_MANAG_FUNC_PARAM_AFTER ":", sizeof(LOGS_MANAG_FUNC_PARAM_AFTER ":") - 1) == 0) {
+            after_s = str2l(&keyword[sizeof(LOGS_MANAG_FUNC_PARAM_AFTER ":") - 1]);
+        }
+        else if(strncmp(keyword, LOGS_MANAG_FUNC_PARAM_BEFORE ":", sizeof(LOGS_MANAG_FUNC_PARAM_BEFORE ":") - 1) == 0) {
+            before_s = str2l(&keyword[sizeof(LOGS_MANAG_FUNC_PARAM_BEFORE ":") - 1]);
+        }
+        else if(strncmp(keyword, LOGS_MANAG_FUNC_PARAM_IF_MODIFIED_SINCE ":", sizeof(LOGS_MANAG_FUNC_PARAM_IF_MODIFIED_SINCE ":") - 1) == 0) {
+            if_modified_since = str2ull(&keyword[sizeof(LOGS_MANAG_FUNC_PARAM_IF_MODIFIED_SINCE ":") - 1], NULL);
+        }
+        else if(strncmp(keyword, LOGS_MANAG_FUNC_PARAM_ANCHOR ":", sizeof(LOGS_MANAG_FUNC_PARAM_ANCHOR ":") - 1) == 0) {
+            anchor = str2ull(&keyword[sizeof(LOGS_MANAG_FUNC_PARAM_ANCHOR ":") - 1], NULL);
+        }
+        else if(strncmp(keyword, LOGS_MANAG_FUNC_PARAM_DIRECTION ":", sizeof(LOGS_MANAG_FUNC_PARAM_DIRECTION ":") - 1) == 0) {
+            direction = !strcasecmp(&keyword[sizeof(LOGS_MANAG_FUNC_PARAM_DIRECTION ":") - 1], "forward") ? 
```

Comment:
Is there a specific reason to accept uppercase here and only accept lower case in other arguments?

Reply:
Yes, I am following the equivalent code in `systemd-journal.plugin`. Not sure what the agreed contract is with the frontend UI, so I suspect the provided `direction` can be either lowercase or uppercase. 

I will keep it as is.